src/WebJobs.Script.WebHost/Diagnostics/KubernetesEventGenerator.cs

```diff
@@ -3,56 +3,83 @@
         {
-            string hostVersion = ScriptHost.Version;
+            JObject metricEvent = new JObject();
+            metricEvent.Add("SubscriptionId", subscriptionId);
+            metricEvent.Add("AppName", appName);
+            metricEvent.Add("FunctionName", functionName);
+            metricEvent.Add("EventName", eventName);
+            metricEvent.Add("Average", average);
+            metricEvent.Add("Minimum", minimum);
+            metricEvent.Add("Maximum", maximum);
+            metricEvent.Add("Count", count);
+            metricEvent.Add("HostVersion", ScriptHost.Version);
+            metricEvent.Add("EventTimeStamp", eventTimestamp.ToString(EventTimestampFormat));
+            metricEvent.Add("Data", NormalizeString(data));
+            metricEvent.Add("RuntimeSiteName", runtimeSiteName);
+            metricEvent.Add("SlotName", slotName);
 
-            _writeEvent($"{subscriptionId},{appName},{functionName},{eventName},{average},{minimum},{maximum},{count},{hostVersion},{eventTimestamp.ToString(EventTimestampFormat)},{NormalizeString(data)},{runtimeSiteName},{slotName},{_podName}");
+            _writeEvent(metricEvent.ToString(Formatting.None));
         }
 
         public override void LogFunctionDetailsEvent(string siteName, string functionName, string inputBindings, string outputBindings, string scriptType, bool isDisabled)
         {
-            _writeEvent($"{siteName},{functionName},{NormalizeString(inputBindings)},{NormalizeString(outputBindings)},{scriptType},{(isDisabled ? 1 : 0)},{_podName}");
         }
 
         public override void LogFunctionExecutionAggregateEvent(string siteName, string functionName, long executionTimeInMs, long functionStartedCount, long functionCompletedCount, long functionFailedCount)
         {
-            _writeEvent($"{siteName},{functionName},{executionTimeInMs},{functionStartedCount},{functionCompletedCount},{functionFailedCount},{_podName}");
         }
 
         public override void LogFunctionExecutionEvent(string executionId, string siteName, int concurrency, string functionName, string invocationId, string executionStage, long executionTimeSpan, bool success)
         {
-            _writeEvent($"{executionId},{siteName},{concurrency},{functionName},{invocationId},{executionStage},{executionTimeSpan},{success},{_podName}");
         }
 
         public override void LogAzureMonitorDiagnosticLogEvent(LogLevel level, string resourceId, string operationName, string category, string regionName, string properties)
         {
-            _writeEvent($"{(int)ToEventLevel(level)},{resourceId},{operationName},{category},{regionName},{NormalizeString(properties.Replace("'", string.Empty))},{DateTime.UtcNow.ToString()},{_podName}");
+            _writeEvent($"{(int)ToEventLevel(level)},{resourceId},{operationName},{category},{regionName},{NormalizeString(properties.Replace("'", string.Empty))},{DateTime.UtcNow.ToString()}");
```

Comment:
Should this be JSON as well? as there is no schema in this environment?

Reply:
This is different from Kusto logs. These logs go to azure monitor, where a schema does need to be enforced.